ux.symfony.com/src/Twig/CookBookList.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Twig;
+
+use App\Model\Recipe;
+use App\Service\CookBookParser;
+use App\Service\CookBookRepository;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
+
+#[AsTwigComponent]
+class CookBookList
+{
+    public function __construct(
+        private CookBookRepository $recipeRepository,
+        private CookBookParser $recipeParser,
+        private UrlGeneratorInterface $urlGenerator,
+    ) {
+    }
+
+    public function getRecipes(): array
+    {
+        $finder = $this->recipeRepository->findAll();
+
+        $recipes = [];
+        foreach ($finder as $file) {
+            $content = $file->getContents();
+
+            $recipes[] = new Recipe(
+                name: $this->recipeParser->getTitle($content),
+                description: $this->recipeParser->getDescriptions($content),
+                route: $this->urlGenerator->generate('app_cookbook_recipe', ['slug' => $file->getBasename('.md')]),
+                image: $this->recipeParser->getImage($content),
+                content: $content,
+                tags: []
+            );
```

Comment:
This is the job of the repository for me.

The url should/could be generated in Twig too, i guess